packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -872,8 +872,12 @@ String getLinuxBuildDirectory([TargetPlatform? targetPlatform]) {
 }
 
 /// Returns the Windows build output directory.
-String getWindowsBuildDirectory() {
-  return globals.fs.path.join(getBuildDirectory(), 'windows');
+String getWindowsBuildDirectory([TargetPlatform? targetPlatform]) {
+  final String arch = (targetPlatform == null) ?
+      _getCurrentHostPlatformArchName() :
+      targetPlatform.simpleName;
+  final String subDirs = 'windows\\$arch';
+  return globals.fs.path.join(getBuildDirectory(), subDirs);
```

Comment:
Can we simplify here?

```suggestion
  return globals.fs.path.join(getBuildDirectory(), 'windows', arch);
```